tools/pipeline_perf_test/load_generator/loadgen.py

```diff
@@ -209,18 +222,144 @@ def worker_thread(self, thread_id: int, args: dict) -> None:
                     self.increment_metric("late_batches")
                 next_send_time += batch_interval
 
+    def syslog_worker_thread(self, thread_id: int, args: dict) -> None:
+        """
+        Worker thread that sends syslog messages to a syslog server.
+        """
+        syslog_server = os.getenv("SYSLOG_SERVER", "localhost")
+        syslog_port = int(os.getenv("SYSLOG_PORT", "514"))
+
+        # Create TCP socket for syslog
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```

Comment:
We want to prioritize testing UDP first.

Reply:
good point. let me send the follow up to pick between tcp and udp.